distributed/client.py

```diff
@@ -648,15 +648,24 @@ def __init__(
                 "Unexpected keyword arguments: {}".format(str(sorted(kwargs)))
             )
 
+        from .deploy import Cluster
+
         if isinstance(address, (rpc, PooledRPCCall)):
             self.scheduler = address
-        elif hasattr(address, "scheduler_address"):
+        elif isinstance(address, Cluster):
```

Comment:
We don't require that downstream implementations of Dask Cluster types inherit from our type.  Perhaps maybe you want something like `isinstance(getattr(address, "scheduler_address", None), str)`?